src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,529 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin-upgrades/contracts/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../libraries/Merkle.sol";
+import "../libraries/EIP1271SignatureUtils.sol";
+import "../permissions/Pausable.sol";
+import "./PaymentCoordinatorStorage.sol";
+
+/**
+ * @title PaymentCoordinator
+ * @author Eigen Labs Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @notice  This is the contract for payments in EigenLayer. The main functionalities of this contract are
+ * - enabling any ERC20 payments from AVSs to their operators and stakers for a given time range
+ * - allowing stakers and operators to claim their earnings including a commission bips for operators
+ * - allowing the protocol to provide ERC20 tokens to stakers over a specified time range
+ */
+contract PaymentCoordinator is
+    Initializable,
+    OwnableUpgradeable,
+    Pausable,
+    ReentrancyGuardUpgradeable,
+    PaymentCoordinatorStorage
+{
+    using SafeERC20 for IERC20;
+
+    /// @dev Index for flag that pauses payForRange payments
+    uint8 internal constant PAUSED_PAY_FOR_RANGE = 0;
+    /// @dev Index for flag that pauses payAllForRange payments
+    uint8 internal constant PAUSED_PAY_ALL_FOR_RANGE = 1;
+    /// @dev Index for flag that pauses
+    uint8 internal constant PAUSED_CLAIM_PAYMENTS = 2;
+    /// @dev Index for flag that pauses submitRoots
+    uint8 internal constant PAUSED_SUBMIT_ROOTS = 3;
+
+    /// @dev Salt for the earner leaf, meant to distinguish from tokenLeaf since they have the same sized data
+    uint8 internal constant EARNER_LEAF_SALT = 0;
+    /// @dev Salt for the token leaf, meant to distinguish from earnerLeaf since they have the same sized data
+    uint8 internal constant TOKEN_LEAF_SALT = 1;
+
+    /// @dev Chain ID at the time of contract deployment
+    uint256 internal immutable ORIGINAL_CHAIN_ID;
```

Comment:
We have some constants/immutables here, and some in `PaymentCoordinatorStorage`. Is there a reason we have these here? Otherwise I'd say we should put them all in one place for consistency.

Reply:
Personally prefer the immutables being in `PaymentCoordinatorStorage` as it has the constructor defining them there (although we could just have the immutables in `PaymentCoordinator` itself).
Cleaned them up with constants into `PaymentCoordinator` and immutables in `PaymentCoordinatorStorage`
https://github.com/Layr-Labs/eigenlayer-contracts/pull/514/commits/90e157c318910a501ca4e4db9c2b6ebe0240d80c